internal/command/user_idp_link.go

```diff
@@ -15,15 +15,22 @@ func (c *Commands) AddUserIDPLink(ctx context.Context, userID, resourceOwner str
 	if userID == "" {
 		return nil, zerrors.ThrowInvalidArgument(nil, "COMMAND-03j8f", "Errors.IDMissing")
 	}
-	if err := c.checkUserExists(ctx, userID, resourceOwner); err != nil {
+
+	existingUser, err := c.userWriteModelByID(ctx, userID, resourceOwner)
+	if err != nil {
 		return nil, err
 	}
+	if !isUserStateExists(existingUser.UserState) {
+		return nil, zerrors.ThrowPreconditionFailed(nil, "COMMAND-vzktar7b7f", "Errors.User.NotFound")
+	}
 	if userID != authz.GetCtxData(ctx).UserID {
-		if err := c.checkPermission(ctx, domain.PermissionUserWrite, resourceOwner, userID); err != nil {
+		if err := c.checkPermission(ctx, domain.PermissionUserWrite, existingUser.ResourceOwner, existingUser.AggregateID); err != nil {
 			return nil, err
 		}
 	}
-	event, err := addLink(ctx, c.eventstore.Filter, user.NewAggregate(userID, resourceOwner), link)
+	// We ignore the deprecation warning here because we still need to support the deprecated field.
```

Comment:
ignoring the deprecation warning for the filter is ok at the moment, but what do you mean by `support the deprecated field`?

Reply:
i now search for that and the only thing i could find was a part where we specifically need access to a deprecated `GetCreationDate()`, which was deprecated in the API and has nothing to do with the deprecation of the filter